### PR TITLE
#356 enabled IT for ReservationSettings

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #356 

* first step: activated IT tests, as the functionality is already implemented in [Reservation views.py](https://github.com/timelesslounge/timelessis/blob/master/timeless/reservations/views.py#L19)
* fix url names and params
* removed puzzle

### CODE REVIEW

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this pull request is too small, just 7 lines changed (less than 10), there will be no formal code review, see [§53](http://www.zerocracy.com/policy.html#53) and [§28](http://www.zerocracy.com/policy.html#28); in the future, try to make sure your pull requests are not too small; @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) please review this and merge or reject

**Comment by emilianodellacasa** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/timelesslounge/timelessis/issues/467#issuecomment-469027636))

@emilianodellacasa Job [#467](https://github.com/timelesslounge/timelessis/issues/467) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#467](https://github.com/timelesslounge/timelessis/issues/467) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/f7978cd3-6fae-4888-930a-d3c656138973); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_reservation_settings_test.py [hunk 1]
```diff
@@ -1,44 +1,33 @@
-"""
-@todo #291:30min Enable all tests below when then functionality in for
- ReservationSettings will be enabled. There will be tuning needed because there
- are no endpoint names in the views.
-"""
 import datetime
 from http import HTTPStatus
-
```
Comment (vryazanov) [inline tests/integration/it_reservation_settings_test.py]: @amihaiemil let's add blank lines as it was before:
```python
import datetime
from http import HTTPStatus

import pytest
from flask import url_for

from timeless.reservations import models
```
it's a pep8rule.

**Comment by amihaiemil** [comment]
@vryazanov I had a look again now, the Travis CI is failing. I think there is more stuff missing than I thought.

File: tests/integration/it_reservation_settings_test.py [hunk 2]
```diff
@@ -48,13 +37,12 @@ def test_edit(client):
     assert html.count(new_data["comment"]) == 1
 
 
-@pytest.mark.skip
 def test_delete(client):
     assert client.get(
-        url_for("reservations.settings_delete", id=1)
+        url_for("reservations.settings_delete", setting_id=1)
```
Comment (vryazanov) [inline tests/integration/it_reservation_settings_test.py]: @amihaiemil looks this line doesn make sense, I think we can remove it. wdyt?

**Comment by amihaiemil** [comment]
@vryazanov can you have a look on Travis, pls? I don't understand where it sees that ``reservations_settings_list`` -- I can't find this String in the source code locally :-?

File: timeless/reservations/views.py [hunk 3]
```diff
@@ -47,7 +47,7 @@ class SettingsCreateView(views.CreateView):
     model = models.ReservationSettings
     template_name = "restaurants/tables/create_edit.html"
     success_view_name = "reservation_settings_list"
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @amihaiemil here should be `reservations.settings_list`, I guess

**Comment by 0crat** [comment]
@vryazanov[/z](https://www.0crat.com/u/vryazanov) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: tests/integration/it_reservation_settings_test.py [hunk 4]
```diff
@@ -1,44 +1,33 @@
...
-@pytest.mark.skip
 def test_edit(client):
     data = reservation_data("nicer comment here")
     new_data = reservation_data("different comment here")
-    client.post(url_for("reservations.settings_create"), data=data)
+    client.post(url_for("reservations.settings_create_view"), data=data)
     identifier = models.ReservationSettings.query.first().id
     client.post(
-        url_for("reservations.settings_create", id=identifier), data=new_data
+        url_for("reservations.settings_create_view", id=identifier), data=new_data
```
Comment (amihaiemil) [inline tests/integration/it_reservation_settings_test.py]: @VladimirRyzhkov Here there should be the url for settings edit. Do you happen to know what that one is? Maybe ``reservations.settings_edit``? I've no idea :(

Comment (vryazanov) [reply tests/integration/it_reservation_settings_test.py]: @amihaiemil  there is no endpoint for updating reservation settings..

**Comment by amihaiemil** [comment]
@vryazanov test_create now works. I skipped the other 3 and left a puzzle for continuing since it was already too much :D

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#467](https://github.com/timelesslounge/timelessis/issues/467) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
The job [#467](https://github.com/timelesslounge/timelessis/issues/467) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**